client/api/api_test.go

```diff
@@ -0,0 +1,230 @@
+package api
+
+import (
+	. "github.com/onsi/gomega"
+	"io"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"testing"
+)
+
+func TestErrorWithNoMessage(t *testing.T) {
+	// assemble
+	goOmegaInstance := NewWithT(t)
+
+	var errWithNoMessage = Error{
+		StatusCode: rand.Int(),
+		Code:       RandStr(),
+		Message:    "",
+		Body:       RandStr(),
+		Header:     http.Header{"": make([]string, 1)},
+		URL:        RandStr(),
+	}
+
+	// act
+	err := errWithNoMessage.Error()
+
+	// assert
+	goOmegaInstance.Expect(err).To(BeAssignableToTypeOf(""))
```

Comment:
similar or exact the same?

Reply:
Identical. The only difference in both case is that the testing function would return a string with different content. Thus the assert is identical, because testing for the content of the message is unstable.